### PR TITLE
fix format duration rounding days error

### LINKED ISSUE #530: Wrong created time
Snímek obrazovky 2020-06-26 v 8 19 41

Some bug is probably in created time. Look at this image there should be something like 19h ago but there is 1 day and 19 hours. Is it bug or could it be some my local issue? I have 3.5.0 version and 1.16 k8s cluster version.

Thank you!

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <smalton@mirantis.com>

Fixes #530.

### CODE REVIEW

**Review by aleksfront — APPROVED** [review]
Looking good!

File: src/renderer/utils/formatDuration.ts [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
+    .filter(([dur, _suf]) => dur > 0)
+    .filter(([_dur, suf], i) => i === 0 || suf !== "s") // remove seconds, unless it is the only one
+    .map(([dur, suf]) => dur + suf);
+
   if (compact) {
-    result = result.split(" ")[0];
-  }
-  if (!result) {
-    return "<1m";
+    return meaningfulValues[0];
```
Comment (jim-docker) [inline src/renderer/utils/formatDuration.ts:31]: If `timeValue` is `0` then `meaningfulValues` will be empty array, right? Maybe you should check for `0` at the start and return `"0s"`? And add a test for `0`...

Comment (Nokel81) [reply src/renderer/utils/formatDuration.ts:31]: Thanks and good catch

File: src/renderer/utils/formatDuration.ts [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
     duration.minutes(),
+    duration.seconds(),
   ];
-  durationValues.forEach((value, index) => {
-    if (value) result += value + suffixes[index] + " ";
-  });
+
+  const meaningfulValues = durationValues
+    .map((a, i): [number, string] => [a, suffixes[i]])
+    .filter(([dur, _suf]) => dur > 0)
```
Comment (jim-docker) [inline src/renderer/utils/formatDuration.ts:22]: doesn't this filter leave you with an empty array when `timeValue` is `0`? And if so how can `meaningfulValues.join(" ")` produce `"0s"` (when `compact` is `false`)?

File: src/renderer/utils/__tests__/formatDuration.test.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    const significant = formatDuration(31 * second, false);
+    expect(significant).toBe("31s");
+  });
+
+  test("zero duration should output something", () => {
+    const zero = formatDuration(0, false);
+    
+    expect(zero).not.toHaveLength(0);
+    expect(zero).toBe("0s");
```
Comment (jim-docker) [inline src/renderer/utils/__tests__/formatDuration.test.ts]: This expect is all you need, no? It implies `zero` does not have zero length.

File: src/renderer/utils/formatDuration.ts (hunk 2 above)
Comment (Nokel81) [reply src/renderer/utils/formatDuration.ts:22]: Thanks for pointing this out. Apparently I completely missed the fact that `make test` was not running my tests.... (I have fixed that now)

File: src/renderer/utils/formatDuration_spec.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    const insignificant = formatDuration(1 * hour + 2 * minute + 31 * second, false);
+    
+    expect(insignificant).not.toContain("s");
+    expect(insignificant).toBe("1h 2m");
+
+    const significant = formatDuration(31 * second, false);
+    expect(significant).toBe("31s");
+  });
+
+  test("zero duration should output something", () => {
```
Comment (jim-docker) [inline src/renderer/utils/formatDuration_spec.ts:47]: you could add a test for a duration > 0 but less than 1s too...

**Review by jim-docker — APPROVED** [review]

**Pull request merged**